src/coreclr/src/gc/unix/cgroup.cpp

```diff
@@ -394,6 +480,110 @@ class CGroup
         return result;
     }
 
+    static bool GetCGroup1CpuLimit(uint32_t *val)
+    {
+        long long quota;
+        long long period;
+
+        quota = ReadCpuCGroupValue(CGROUP1_CFS_QUOTA_FILENAME);
+        if (quota <= 0)
+            return false;
+
+        period = ReadCpuCGroupValue(CGROUP1_CFS_PERIOD_FILENAME);
+        if (period <= 0)
+            return false;
+
+        ComputeCpuLimit(quota, period, val);
+
+        return true;
+    }
+
+    static bool GetCGroup2CpuLimit(uint32_t *val)
+    {
+        FILE *file = nullptr;
+        char *endptr = nullptr;
+        char *max_quota_string = nullptr;
+        char *period_string = nullptr;
+        char *space = nullptr;
+
+        const size_t lineLen = 1 /* space */ +
+                               (2 * 3 * sizeof(long long)) /* max space for 2 long longs */ +
+                               1 /* newline */ +
+                               1 /* null */;
+        char line[lineLen];
+        int sscanRet = 0;
+        long long quota = 0;
+        long long period = 0;
+
+        bool result = false;
+
+        if (s_cpu_cgroup_path == nullptr)
+            return false;
+
+        char filename[PATH_MAX];
+        snprintf(filename, PATH_MAX, "%s%s", s_cpu_cgroup_path, CGROUP2_CPU_MAX_FILENAME);
+        file = fopen(filename, "r");
+        if (file == nullptr)
+            goto done;
+
+        if (fgets(line, lineLen, file) == nullptr)
+            goto done;
+
+        // The expected format is:
+        //     $MAX $PERIOD
+        // Where "$MAX" may be the string literal "max"
+
+        space = (char*) memchr(line, ' ', lineLen);
```

Comment:
A nit - it seems that using strtok to parse the line would result in an easier to read code.
```c++
char* context = nullptr;
max_quota_string = strtok_r(line, " ", &context);
if (max_quota_string == nullptr)
{
   ...
}
period_string = strtok_r(nullptr, " ", &context);
if (period_string == nullptr)
{
   ...
}

```

Reply:
Thanks. Fixed now.